base/common/src/main/java/org/dogtagpki/tps/TPSConnection.java

```diff
@@ -62,16 +62,16 @@ public TPSMessage read() throws IOException {
 
         // Check first two bytes from InputStream (s=).
         // The first char can be anything.
-        if((b = in.read()) < 0)
-            throw new IOException("Unexpected end of stream");
-        else
+        if((b = in.read()) >= 0)
             sb.append((char)b);
+        else
+            throw new IOException("Unexpected end of stream");
 
         // The second char must be '='.
-        if((b = in.read()) != '=')
-            throw new IOException("Unexpected end of stream");
-        else
+        if((b = in.read()) == '=')
             sb.append((char)b);
+        else
+            throw new IOException("Unexpected end of stream");
```

Comment:
I think we want to keep the original if-clauses since they throw an exception, but remove the else-clause instead.

Reply:
Ack, I reverted and dropped the `else`s